x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/notification/email/support/EmailServer.java

```diff
@@ -61,12 +67,33 @@ public void deliver(String from, String recipient, InputStream data) throws IOEx
                     throw new RuntimeException("could not create mime message", me);
                 }
             }
-        }), new EasyAuthenticationHandlerFactory((user, passwd) -> {
+        });
+        final EasyAuthenticationHandlerFactory authentication = new EasyAuthenticationHandlerFactory((user, passwd) -> {
             assertThat(user, is(USERNAME));
             assertThat(passwd, is(PASSWORD));
-        }));
+        });
+        server = new SMTPServer(listener, authentication) {
+            @Override
+            public SSLSocket createSSLSocket(Socket socket) throws IOException {
+                if (sslContext == null) {
+                    return super.createSSLSocket(socket);
+                } else {
+                    SSLSocketFactory factory = sslContext.getSocketFactory();
+                    InetSocketAddress remoteAddress = (InetSocketAddress) socket.getRemoteSocketAddress();
+                    SSLSocket sslSocket = (SSLSocket) factory.createSocket(socket, remoteAddress.getHostString(), socket.getPort(), true);
+                    sslSocket.setUseClientMode(false);
+                    sslSocket.setEnabledCipherSuites(sslSocket.getSupportedCipherSuites());
+                    return sslSocket;
+                }
+            }
+        };
         server.setHostName(host);
         server.setPort(0);
+        if (sslContext != null) {
+            server.setEnableTLS(true);
+            server.setRequireTLS(true);
+            server.setHideTLS(false);
```

Comment:
feel-free-to-ignore drive-by nit: are these settings always guaranteed for servers in the wild or is worth randomizing? (not the first one obviously)

Reply:
Fair question. It might be worth having a separate test that randomises them, but for the purposes of this test, I think it is helpful to leave them as is.

Subethamail only supports StartTLS, and not a dedicated TLS port (there are forks on subetha that have added dedicared TLS).
StartTLS intiates a non-TLS connection and then uses an application level protocol to upgrade the connection to TLS. The problem (in general, and in this test) is that it makes it hard to know whether the connection was actually secured with TLS. Since this test is intended to verify the client behaviour when using TLS, we want the server to enforce that TLS is always used so that we don't accidentally pass the test by falling back to a non-TLS connection. And if we require TLS then we shouldn't hide it, because it's unreasonable for a client to know that it should initiate StartTLS if the server doesn't advertise that it supports it (I haven't tested, but I expect it would fail).